starlite/utils/serialization.py

```diff
@@ -31,22 +34,36 @@
     # pydantic specific types
     BaseModel: lambda m: m.dict(),
     ByteSize: lambda b: b.real,
-    EmailStr: str,
     NameEmail: str,
     Color: str,
-    AnyUrl: str,
     SecretField: str,
-    ConstrainedInt: int,
-    ConstrainedFloat: float,
-    ConstrainedStr: str,
     ConstrainedBytes: lambda b: b.decode("utf-8"),
-    ConstrainedList: list,
-    ConstrainedSet: set,
-    ConstrainedFrozenSet: frozenset,
```

Comment:
When would a user ever end up with an instance of one of these pydantic types? AFAICT these types are for type annotations only, during decode I don't ever see pydantic returning concrete instances of them:

```python
In [1]: import pydantic

In [2]: class Test(pydantic.BaseModel):
   ...:     x: pydantic.constr(min_length=3)
   ...: 

In [3]: type(Test(x="12345").x)  # x is a str, not a ConstrainedStr
Out[3]: str

In [4]: type(Test.parse_raw(b'{"x": "12345"}').x)
Out[4]: str
```

Reply:
> When would a user ever end up with an instance of one of these pydantic types?

This can only happen if the constrained value is being instantiated directly. I admit it's an edge case, and one could argue that it shouldn't be done, but I've seen it a few times in the wild.